### PR TITLE
feat(navigation): add support for `redirect` checkpoint

### PR DESCRIPTION
**Pull request by trieloff**
if the visitor arrived on the page via a (server-side) redirect, then
fire a `redirect` checkpoint with the original URL (if CDN supports it)
and the number of redirect hops

see https://cq-dev.slack.com/archives/C07198KKQ07/p1725446009159669

### CODE REVIEW

File: modules/index.js [hunk 1]
```diff
@@ -122,10 +122,20 @@ function addNavigationTracking() {
     } else {
       sampleRUM('enter', payload); // enter site
     }
+    fflags.enabled('redirect', () => {
+      const from = new URLSearchParams(window.location.search).get('redirect-from');
```
Comment (kptdobe) [inline modules/index.js:126]: See slack for discussion on this ugly search param

File: test/it/navigation.test.html [hunk 2]
```diff
@@ -28,7 +28,8 @@
             getEntries() {
               return [
                 {
-                  type: 'navigate'
+                  type: 'navigate',
```
Comment (kptdobe) [inline test/it/navigation.test.html:31]: Missing test for the "ugly search param" use case.

**Review by kptdobe — CHANGES_REQUESTED** [review]

File: test/it/navigation.test.html (hunk 2 above)
Comment (trieloff) [reply test/it/navigation.test.html:31]: Test is now included.

File: modules/index.js [hunk 3]
```diff
@@ -122,10 +122,20 @@ function addNavigationTracking() {
...
+      if (redirectCount || from) {
+        sampleRUM('redirect', { source: from, target: redirectCount || 1 });
```
Comment (kptdobe) [inline modules/index.js:128]:
```suggestion
        sampleRUM('redirect', { source: from || '', target: redirectCount || 1 });
```

`from` might be null

**Review by kptdobe — APPROVED** [review]

**Pull request merged**